apps/website/src/app/atoms/icons/TelegramIcon.tsx

```diff
@@ -1,16 +1,19 @@
-export const TelegramIcon = () => (
+import { IconProps } from "@/app/atoms/icons/IconProps";
+
+export const TelegramIcon = (props: IconProps) => (
   <svg
     xmlns="http://www.w3.org/2000/svg"
     width="24"
     height="24"
     viewBox="0 0 24 24"
     fill="none"
-    className="group"
+    className={props.svgClass}
+    {...props.other}
   >
     <path
       d="M21.9597 4.33581C21.9435 4.26415 21.9081 4.19792 21.857 4.14382C21.806 4.08972 21.7411 4.04967 21.6689 4.02773C21.4062 3.97723 21.1343 3.99594 20.8815 4.08191C20.8815 4.08191 3.35842 10.1941 2.35765 10.8709C2.14267 11.0165 2.07017 11.1014 2.03434 11.2008C1.86102 11.686 2.40015 11.8946 2.40015 11.8946L6.91653 13.3226C6.9929 13.3359 7.07142 13.3315 7.14568 13.3097C8.17228 12.6798 17.4784 6.97509 18.0192 6.78345C18.1025 6.75919 18.1666 6.78345 18.15 6.84409C17.935 7.57588 9.89301 14.508 9.84884 14.5501C9.82733 14.5672 9.81057 14.5892 9.80009 14.6142C9.78961 14.6392 9.78575 14.6664 9.78885 14.6932L9.36721 18.9723C9.36721 18.9723 9.19055 20.3041 10.563 18.9723C11.5362 18.0271 12.4703 17.2443 12.937 16.8635C14.4902 17.9042 16.1609 19.0548 16.8817 19.6572C17.003 19.7711 17.1466 19.8601 17.3041 19.9189C17.4616 19.9776 17.6297 20.005 17.7983 19.9993C18.006 19.9747 18.201 19.8894 18.3575 19.7546C18.5139 19.6198 18.6244 19.442 18.6741 19.2448C18.6741 19.2448 21.8656 6.77375 21.9722 5.10317C21.9831 4.94145 21.9972 4.83472 21.9989 4.72232C22.0041 4.59235 21.9909 4.46231 21.9597 4.33581Z"
       fill="#AFAFAF"
-      className="gt_mobile:group-hover:fill-black transition-all duration-200"
+      className={props.pathClass}
```

Comment:
Why do we need two different classes here? You can use `fill-current` and attach the hover styles to the SVG itself.

Reply:
Adjusted for all Footer icons